apps/wakunode2/app.nim

```diff
@@ -512,6 +512,46 @@ proc setupProtocols(node: WakuNode, conf: WakuNodeConf,
       pubsubTopics = conf.topicsDeprecated.split(" ")
     else:
       pubsubTopics = conf.topics
+
+    # Validate topics and update a node's ENR accordingly
+    let parsed_topics = pubsubTopics.mapIt(NsPubsubTopic.parse(it))
+
+    var prevTopic: NsPubsubTopic
+    for i, topic in parsed_topics:
+      if topic.isErr():
+        return err("Pubsub topic parsing error")
+      
+      if i == 0:
+        prevTopic = topic.value
+        continue
+      
+      if prevTopic.kind != topic.value.kind:
```

Comment:
Indeed, this would require iterating twice, but given that it improves readability quite a bit, I think it's worth it.
The 99% use case would be nodes not using any static shards. I'd also suggest extracting most of this logic into a separate unit-testable function, perhaps taking the `parsedTopics` as argument and returning a `Result[RelayShards]`. Then you could do something like:

```
if parsedTopics.anyIt(it.kind == NsPubsubTopicKind.StaticSharding):
  let relayShardsRes = parsedTopics.getRelayShards()
  ## error handling here in case result is err
```

The `getRelayShards` func could then check that `allIt` is StaticSharding and of the same cluster before returning the value.

Reply:
Is there a way to handle error when using functional style Nim?
```
if parsedTopicsRes.anyIt(it.isErr()):
    return err(it.error)
```
Something like that?